src/cript/nodes/primary_nodes/project.py

```diff
@@ -76,6 +80,37 @@ def __init__(
         self._json_attrs = replace(self._json_attrs, name=name, collections=collections, material=material)
         self.validate()
 
+    def validate(self):
+        # First validate like other nodes
+        super().validate()
+
+        # Check graph for orphraned nodes, that should be listed in project
+        # Project.materials should contain all material nodes
+        all_materials = self.find_children({"node": ["Material"]})
+        for material in all_materials:
```

Comment:
very picky small improvement, I would just name it as `for material in project_materials:`

Reply:
sure, np.